template.lua

```diff
@@ -121,4 +121,30 @@ function PLUGIN:EnvKeys(ctx)
             value = mainPath .. "/bin"
         }
     }
-end
\ No newline at end of file
+end
+
+--- When user invoke `use` command, this function will be called to get the 
+--- valid version information.
+--- @param ctx table Context information
+function PLUGIN:PreUse(ctx)
+    local runtimeVersion = ctx.runtimeVersion
+    --- user input version
+    local version = ctx.version
```

Comment:
From what I can think of so far, I think there is one missing parameter, `previousVersion`, which represents the version that was in use before the current switch. If there was no previous version, this parameter will be nil.

The main purpose of this is to allow for subsequent operations that may need to run tools, such as Redis or Tomcat. This way, previous versions can be shut down here.